Tests/RequestTests.swift

```diff
@@ -0,0 +1,212 @@
+//
+//  RequestTests.swift
+//  HyperspaceTests
+//
+//  Created by Tyler Milner on 6/26/17.
+//  Copyright © 2017 Bottle Rocket Studios. All rights reserved.
+//
+
+import XCTest
+import Hyperspace
+
+class RequestTests: XCTestCase {
+    
+    // MARK: - Tests
+    
+    func test_SimpleGETRequestWithoutQueryParametersOrHeaders_GeneratesCorrectURLRequest() {
+        let request: Request<String, AnyError> = .simpleGET
+        assertParameters(for: request)
+    }
+    
+    func test_SimpleGETRequestWithHeaders_GeneratesCorrectURLRequest() {
+        var request: Request<String, AnyError> = .simpleGET
+        request.headers = [.contentType: .applicationJSON]
+        
+        assertParameters(headers: ["Content-Type": "application/json"], for: request)
+    }
+    
+    func test_SimplePOSTRequestWithData_GeneratesCorrectURLRequest() {
+        let bodyData = "Test".data(using: .utf8)!
+
+        var request: Request<String, AnyError> = .simplePOST
+        request.body = HTTP.Body(bodyData)
+        
+        assertParameters(method: "POST", body: bodyData, for: request)
+    }
+
+    func test_RequestWithoutExplicitCachePolicyAndTimeout_ReturnsDefaultCachePolicyAndTimeout() {
+        let timeout: TimeInterval = 1
+        let cachePolicy: URLRequest.CachePolicy = .returnCacheDataDontLoad
+        
+        RequestDefaults.defaultTimeout = timeout
+        RequestDefaults.defaultCachePolicy = cachePolicy
+              
+        let request: Request<EmptyResponse, AnyError> = .cachePolicyAndTimeoutRequest
+        XCTAssert(request.cachePolicy == cachePolicy)
+        XCTAssert(request.timeout == timeout)
```

Comment:
Not sure why `XCTAssertEqual` was not used here.
```suggestion
        XCTAssertEqual(request.cachePolicy, cachePolicy)
        XCTAssertEqual(request.timeout, timeout)
```